src/resources/tcp_bsd.cc

```diff
@@ -340,7 +340,7 @@ static Object* get_address(int id, Process* process, bool peer) {
   if (result != 0) return Primitive::os_error(errno, process);
   char buffer[16];
   uint32_t addr_word = ntohl(sin.sin_addr.s_addr);
-  sprintf(buffer, "%d.%d.%d.%d",
+  snprintf(buffer, sizeof(buffer), "%d.%d.%d.%d",
```

Comment:
I would add a
`buffer[sizeof(buffer) - 1] = '\0'` after this call.

Reply:
Done.